### PR TITLE
feat: allows to define rules that are enabled when a given value is a partial match to a target attribute

### PR DESCRIPTION
**Pull request by 3846masa**
Fix https://github.com/bucketeer-io/bucketeer/issues/1188

**This PR is a feature request.**

In Bucketeer rules, there are `contains` (= `Clause.Operator.IN`), which is essentially the same behavior as `Clause.Operator.PARTIALLY_MATCH`.

This PR defines an Operator that can specify a condition on whether a user's attribute partially matches a given value.

---

This pull request introduces a new clause operator, `PARTIALLY_EQUALS`, and integrates it across the codebase. The changes span multiple files, including Go source files, protocol buffer definitions, and UI components.

### New Clause Operator Implementation:

* [`evaluation/clause_evaluator.go`](diffhunk://#diff-8ecba8495933fb25c806ea83516e46b8e47620f91cb1397b47d2d8ce2c211d81R64-R65): Added a new case `featureproto.Clause_PARTIALLY_EQUALS` in the `Evaluate` method and implemented the `partiallyEquals` function to check if the target value contains any of the specified values. [[1]](diffhunk://#diff-8ecba8495933fb25c806ea83516e46b8e47620f91cb1397b47d2d8ce2c211d81R64-R65) [[2]](diffhunk://#diff-8ecba8495933fb25c806ea83516e46b8e47620f91cb1397b47d2d8ce2c211d81R79-R87)

### Protocol Buffer Updates:

* [`proto/feature/clause.pb.go`](diffhunk://#diff-dc940c3d289d918bd6c4accfb5c2d2ed2231066a161a5111ecdb56d3c009f44cR53): Introduced `Clause_PARTIALLY_EQUALS` with value `12` and updated the corresponding maps and descriptors to include this new operator. [[1]](diffhunk://#diff-dc940c3d289d918bd6c4accfb5c2d2ed2231066a161a5111ecdb56d3c009f44cR53) [[2]](diffhunk://#diff-dc940c3d289d918bd6c4accfb5c2d2ed2231066a161a5111ecdb56d3c009f44cR71) [[3]](diffhunk://#diff-dc940c3d289d918bd6c4accfb5c2d2ed2231066a161a5111ecdb56d3c009f44cR86) [[4]](diffhunk://#diff-dc940c3d289d918bd6c4accfb5c2d2ed2231066a161a5111ecdb56d3c009f44cL191-R194) [[5]](diffhunk://#diff-dc940c3d289d918bd6c4accfb5c2d2ed2231066a161a5111ecdb56d3c009f44cL200-R203) [[6]](diffhunk://#diff-dc940c3d289d918bd6c4accfb5c2d2ed2231066a161a5111ecdb56d3c009f44cL211-R219)
* [`proto/feature/clause.proto`](diffhunk://#diff-00f2cc08ab5346e3e32498f514aa88819b7af7917cf664d318d787031b344349R35): Added `PARTIALLY_EQUALS` to the `Clause` enum.

### UI Integration:

* [`ui/web-v2/src/assets/lang/en.json`](diffhunk://#diff-8ce19346eb9f703377de8f86c66b260b6bee157d5e3d7ba0eb791e6be447d06eR290): Added the English translation for the `partiallyEqual` operator.
* [`ui/web-v2/src/assets/lang/ja.json`](diffhunk://#diff-81cd4639cc10aeda280257a420a2ae4d7a8c7a0e0d5546ca0bbb7534690d3a2bL287-R290): Added the Japanese translation for the `partiallyEqual` operator.
* [`ui/web-v2/src/components/FeatureTargetingForm/index.tsx`](diffhunk://#diff-1212e7c54dadfe75f84013af7a1f67c75e2f5c52699077d9b55fb65a3c16ff7aR942-R945): Included the `PARTIALLY_EQUALS` operator in the clause comparison operator options.
* [`ui/web-v2/src/lang/messages.ts`](diffhunk://#diff-87c968f9963cee1754affbebe2dd0394897025d855d8304b12c03e20c49c147fR1727-R1730): Defined the message for the `partiallyEqual` operator.
* [`ui/web-v2/src/proto/feature/clause_pb.js`](diffhunk://#diff-435849b9160cd67a0b9109aa6e5de87290b69e3a66529066a8244946bda9f5a5L221-R222): Updated the `Clause.Operator` object to include `PARTIALLY_EQUALS`.

### CODE REVIEW

File: proto/feature/clause.proto [hunk 1]
```diff
@@ -32,6 +32,7 @@ message Clause {
     AFTER = 10;
     // Attribute is feature ID, and value is variation ID.
     FEATURE_FLAG = 11;
+    PARTIALLY_EQUALS = 12;
```
Comment (cre8ivejp) [inline proto/feature/clause.proto:35]: Thank you so much for your contribution!

The `PARTIALLY_MATCH` is a better fit for this case.
Could you update the other places?

```suggestion
    PARTIALLY_MATCH = 12;
```

Comment (3846masa) [reply proto/feature/clause.proto]: @cre8ivejp 
I recreated the commit to have `PARTIALLY_MATCH`.

**Comment by cre8ivejp** [comment]
@3846masa, to fix the CI issues, you must update the generated proto files using the command `make proto-all` at the project's root and push the changes.
Are you working in the [GitHub Dev Container](https://github.com/bucketeer-io/bucketeer/blob/main/DEVELOPMENT.md#set-up-the-dev-container)? If so, you just need to run the command above because the container already contains all the necessary tools. Otherwise, you will need to install the tools on your local machine.

**Comment by 3846masa** [comment]
@cre8ivejp 
I executed `make proto-all` in VSCode Dev Container.

**Review by kentakozuka — APPROVED** [review]
Thanks! Amazing improvement 🎉

**Comment by cre8ivejp** [comment]
@3846masa, please rebase your branch to fix the conflict, so we can merge it.

**Comment by 3846masa** [comment]
@cre8ivejp I was rebased and fixed conflict.

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**